### ISSUE TITLE
Better autotracking support?

### ISSUE THREAD

**Issue by andreyfel** [poteto/ember-changeset#492]
I see that this addon tries to support autotracking but there is a mixed approach:
`@tracked _changes` and notifyPropertyChange.
Perhaps we should use TrackedObject from https://github.com/pzuraq/tracked-built-ins in order to properly support autotracking?

**Comment by andreyfel** [comment]
I see lack of autotracking support when I define a getter on a changeset which needs to be invalidated when one of the underlying root state properties changes.

**Comment by snewcomer** [comment]
We explicitly reset changes when we know about them, thus allowing us to use plain old `@tracked`.  Is state changing outside of the library's control?

https://github.com/validated-changeset/validated-changeset/blob/1b8265cd9a37ab2bf9e93b021f4dc828e0bd2b39/src/index.ts#L774

**Comment by andreyfel** [comment]
No, it is not outside. Will try to make a test for the issue I'm experiencing.

**Comment by andreyfel** [comment]
#495 illustrates the issue

**Comment by snewcomer** [comment]
ref #455

**Comment by andreyfel** [comment]
I think the uniqueness of my use case is validating a derived state vs root state validated in #456.
